systest/cluster/nodes.go

```diff
@@ -185,6 +184,8 @@ func deployNodes(ctx *testcontext.Context, name string, from, to int, flags []De
 		i := i
 		eg.Go(func() error {
 			setname := fmt.Sprintf("%s-%d", name, i)
+			idx := i % ctx.PoetSize
+			flags = append(flags, PoetEndpoint(poetEndpoint(idx)))
```

Comment:
this will extend flags on every iteration, and potentially will even modify original flags array

Reply:
copied it out. thanks for catching this

Reply:
this is not safe. if original flags slice has a capacity it will be extended with new flag value, and the next one will overwrite it

```
a := make([]string, 1, 2)
a[0] = "1"
b := a
b = append(b, "2")
c := a
c = append(c, "3")
fmt.Println(b, c)
// [1, 3], [1,3]
```

Reply:
sorry. bad with slices. fixed.